src/main/java/seedu/address/ui/ModuleListPanel2.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.ui;
+
+import java.util.logging.Logger;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.events.ui.ModulePanelSelectionChangedEvent;
+import seedu.address.model.module.Module;
+
+
+/**
+ * Describes the design of second UI panel containing module list display.
+ */
+public class ModuleListPanel2 extends UiPart<Region> {
+    private static final String FXML = "ModuleListPanel2.fxml";
+    private final Logger logger = LogsCenter.getLogger(ModuleListPanel2.class);
+
+    @FXML
+    private ListView<Module> moduleListView;
+
+    public ModuleListPanel2(ObservableList<Module> moduleList) {
+        super(FXML);
+        setConnections(moduleList);
+        registerAsAnEventHandler(this);
+    }
+
+    private void setConnections(ObservableList<Module> moduleList) {
+        moduleListView.setItems(moduleList);
+        moduleListView.setCellFactory(listView -> new ModuleListViewCell());
+        setEventHandlerForSelectionChangeEvent();
+    }
+
+    private void setEventHandlerForSelectionChangeEvent() {
+        moduleListView.getSelectionModel().selectedItemProperty()
+                .addListener((observable, oldValue, newValue) -> {
+                    if (newValue != null) {
+                        logger.fine("Selection in module list panel changed to : '" + newValue + "'");
+                        raise(new ModulePanelSelectionChangedEvent(newValue));
+                    }
+                });
+    }
+
+    /**
+     * Scrolls to the {@code ModuleCard} at the {@code index} and selects it.
+     */
+    //private void scrollTo(int index) {
```

Comment:
whats the reason this is commented out?

Reply:
Not really related, but you can actually comment out large chunk of code using `/* large chunk of code /*`
```
/*
private void scrollTo(int index) {
  Platform.runLater(() -> {
    moduleListView.scrollTo(index);
    moduleListView.getSelectionModel().clearAndSelect(index);
  });
}

@Subscribe
private void handleJumpToListRequestEvent(JumpToListRequestEvent event) {
  logger.info(LogsCenter.getEventHandlingLogMessage(event));
  scrollTo(event.targetIndex);
}
*/
```

Reply:
Ohhh haha thanks @alexkmj ! And @jeremiah-ang we don't use indexing in our modules when we display them, so I didn't think we needed it.